### ISSUE TITLE
Bad Table Name in Select

### ISSUE THREAD

**Issue by tuggernuts** [efcore/EFCore.NamingConventions#147]
I have been using EF Core nightly builds, and just updated to latest. In doing so I found a break, where the table is not put in snake case.

As of 7.0.0-preview.7.22358.1, a bad (not formatted) table name is included in a join involving inheritance:

```SQL
SELECT [t2].[id], [t2].[alias], [t2].[campus_id], [t2].[cluster_id], [t2].[country_id], [t2].[created_by], [t2].[created_on], [t2].[deleted_by], [t2].[deleted_on], [t2].[name], [t2].[owner_id], [t2].[type_id], [t2].[updated_by], [t2].[updated_on], [t2].[web_site], [t2].[affiliation_id], [t2].[affiliation_id0], [t2].[band_id], [t2].[city_id], [t2].[fcc_facility_id], [t2].[format_id], [t2].[frequency], [t2].[primary_market_id], [t2].[secondary_metro_id], [t2].[has_affiliates], [t2].[affiliation_id1], [t2].[affiliation_id2], [t2].[channel], [t2].[city_id0], [t2].[fcc_facility_id0], [t2].[market_id], [t2].[Discriminator], [t].[id], [t].[id0], [t].[id1], [t].[company_id], [t2].[id0], [t2].[content], [t2].[id1], [t2].[geolocation_latitude], [t2].[geolocation_longitude], [t2].[id2], [t2].[geolocation_latitude0], [t2].[geolocation_longitude0]
      FROM (
          SELECT TOP(1) [c].[id], [c0].[id] AS [id0], [i].[id] AS [id1], [c1].[company_id]
          FROM [companies] AS [c]
          INNER JOIN [countries] AS [c0] ON [c].[country_id] = [c0].[id]
          INNER JOIN [industries] AS [i] ON [c].[industry_id] = [i].[id]
          LEFT JOIN [company_notes] AS [c1] ON [c].[id] = [c1].[company_id]
          WHERE [c].[id] = @__id_0
      ) AS [t]
      INNER JOIN (
          SELECT [m].[id], [m].[alias], [m].[campus_id], [m].[cluster_id], [m].[country_id], [m].[created_by], [m].[created_on], [m].[deleted_by], [m].[deleted_on], [m].[name], [m].[owner_id], [m].[type_id], [m].[updated_by], [m].[updated_on], [m].[web_site], [r0].[affiliation_id], [r1].[affiliation_id] AS [affiliation_id0], [r1].[band_id], [r1].[city_id], [r1].[fcc_facility_id], [r1].[format_id], [r1].[frequency], [r1].[primary_market_id], [r1].[secondary_metro_id], [t0].[has_affiliates], [t1].[affiliation_id] AS [affiliation_id1], [t3].[affiliation_id] AS [affiliation_id2], [t3].[channel], [t3].[city_id] AS [city_id0], [t3].[fcc_facility_id] AS [fcc_facility_id0], [t3].[market_id], CASE
              WHEN [t3].[id] IS NOT NULL THEN N'TvStation'
              WHEN [t1].[id] IS NOT NULL THEN N'TvShow'
              WHEN [t0].[id] IS NOT NULL THEN N'TvNetwork'
              WHEN [r1].[id] IS NOT NULL THEN N'RadioStation'
              WHEN [r0].[id] IS NOT NULL THEN N'RadioProgram'
              WHEN [r].[id] IS NOT NULL THEN N'RadioNetwork'
          END AS [Discriminator], [p].[id] AS [id0], [p].[content], [r1].[id] AS [id1], [r1].[geolocation_latitude], [r1].[geolocation_longitude], [t3].[id] AS [id2], [t3].[geolocation_latitude] AS [geolocation_latitude0], [t3].[geolocation_longitude] AS [geolocation_longitude0]
          FROM [MediaProperties] AS [m]
          LEFT JOIN [radio_networks] AS [r] ON [m].[id] = [r].[id]
          LEFT JOIN [radio_programs] AS [r0] ON [m].[id] = [r0].[id]
          LEFT JOIN [radio_stations] AS [r1] ON [m].[id] = [r1].[id]
          LEFT JOIN [tv_networks] AS [t0] ON [m].[id] = [t0].[id]
          LEFT JOIN [tv_shows] AS [t1] ON [m].[id] = [t1].[id]
          LEFT JOIN [tv_stations] AS [t3] ON [m].[id] = [t3].[id]
          LEFT JOIN [property_notes] AS [p] ON [m].[id] = [p].[id]
      ) AS [t2] ON [t].[id] = [t2].[owner_id]
      ORDER BY [t].[id], [t].[id0], [t].[id1], [t].[company_id], [t2].[id], [t2].[id0]
```
Note that `FROM [MediaProperties] AS [m]` should be `FROM [media_properties] AS [m]` 

This works as expected up to & including 7.0.0-preview.7.22357.2.

**Comment by roji** [comment]
Any chance you can put together a minimal code sample which shows this erroring? That would help investigating this...

**Comment by tuggernuts** [comment]
> Any chance you can put together a minimal code sample which shows this erroring? That would help investigating this...

I understand. I'll see what I can do.

**Comment by tuggernuts** [comment]
Hi @roji.

I was able to create a relatively simple sample app which recreates the issue, located [here](https://github.com/tuggernuts/EfCore7.NamingIssue). It's a simple console app w/ a SQLite in-memory database, and I'm dumping the log to the Debug console.

You'll find a company entity, with a list of aircraft assets. There is a base aircraft type, with several inheritors.

Beginning with 7.0.0-preview.7.22358.1, you will see that when EF Core pulls in the associated aircraft assets, it looks for a table named `Aircraft` instead of the properly snake-cased `aircraft`. 

Since this is SQLite, there's no SQL exception due to SQLite's case-insensitivity -- but the issue is still visible.

I know this is a bit of an edge case, and I don't need to be on the latest builds. But hopefully you'll find this helpful down the road...

**Comment by roji** [comment]
Thanks, it's definitely important to know if some change happened in EF which causes something like this. Unfortunately I'm very busy with other stuff at the moment, so it may take a bit of time before I can look into this...

**Comment by tuggernuts** [comment]
Hi, @roji 

I noticed this repo was updated to 7.0.0. 

It seems 7.0.0 is incompatibile with efcore 7.0.0-preview.7.22357.2. But I can't upgrade efcore beyond 22357.2, because this issue(#147) still exists. 

Do you have any idea when you may get a chance to look into this issue? Or maybe point me at a workaround?

Thanks,
Jerry

**Comment by tuggernuts** [comment]
I am able to get around this using a DbCommandInterceptor. Kind of a brute hack, but it works.

**Comment by roji** [comment]
Confirmed this as a duplicate of #184. In the latest bits (about to be released as 8.0.0), I can see `aircraft` in the repo instead of `Aircraft`.

**Comment by roji** [comment]
Duplicate of #184
